### PR TITLE
Fixes #175 by removing hardcoded configuration for LibraryExporter.

### LINKED ISSUE #175: Test failure under Release configuration
The issue here is that the library exporter is hard coded to use the `Debug` configuration https://github.com/aspnet/Scaffolding/blob/27f910e111e3c176acdf904079053d2b9264cbf3/src/Microsoft.Extensions.CodeGeneration.Utils/DotNet/LibraryExporter.cs#L29. 

On our CI, we test using the `Release` configuration which will cause failures when the test try to look for libraries built with `Debug` e.g. " Could not find a part of the path 'C:\gh\Scaffolding\test\TestApps\ModelTypesLocatorTestClassLibrary\bin\ __Debug__ \dnxcore50\ModelTypesLocatorTestClassLibrary.dll'"

I made a temporary fix in https://github.com/aspnet/Scaffolding/commit/243fd3c37c158ec096dcc25e7322e5fbb032b564 to unblock Universe Test but I'm not sure if it's the right fix.

cc @dougbu @kichalla

### PR DESCRIPTION
**Pull request by prafullbhosale**
cc @JunTaoLuo, @abpiskunov

### CODE REVIEW

File: test/Microsoft.Extensions.CodeGeneration.Core.FunctionalTest/TestHelper.cs [hunk 1]
```diff
@@ -19,8 +19,11 @@ public static class TestHelper
     {
         public static IServiceProvider CreateServices(string testAppName)
         {
-            var appEnvironment = new ApplicationEnvironment("TestApp",Directory.GetCurrentDirectory());
-
+#if RELEASE
+            var appEnvironment = new ApplicationEnvironment("TestApp",Directory.GetCurrentDirectory(), "Release");
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.CodeGeneration.Core.FunctionalTest/TestHelper.cs]: nit: space after comma

**Comment by JunTaoLuo** [comment]
Changes looks good, have you tried testing with the environment variable `Configuration` set to `Release` and `Debug` to see if all the tests pass?

**Comment by prafullbhosale** [comment]
Yes all tests pass with both configurations.

File: src/Microsoft.Extensions.CodeGeneration/Program.cs [hunk 2]
```diff
@@ -41,11 +41,18 @@ public static void Main(string[] args)
...
             
             app.OnExecute(() =>
             {
                 var serviceProvider = new ServiceProvider();
                 var context = CreateProjectContext(projectPath.Value());
+                
+                var configuration = appConfiguration.Value()??"Debug";
+                if(configuration != null && !configuration.Equals("Release") && !configuration.Equals("Debug")) 
+                {
+                    throw new ArgumentException($"Invalid value for configuration: {configuration}. "+appConfiguration.Description);
```
Comment (JunTaoLuo) [inline src/Microsoft.Extensions.CodeGeneration/Program.cs]: What's up with the mix of string interpolation and concatenation?

File: src/Microsoft.Extensions.CodeGeneration/Program.cs [hunk 3]
```diff
@@ -41,11 +41,18 @@ public static void Main(string[] args)
...
             var projectPath = app.Option("-p|--project", "Path to project.json", CommandOptionType.SingleValue);
             var packagesPath = app.Option("-n|--nugetPackageDir", "Path to check for Nuget packages", CommandOptionType.SingleValue);
+            var appConfiguration = app.Option("-c|--configuration", "Configuration for the project (Possible values: Debug/ Release)", CommandOptionType.SingleValue);
             
             app.OnExecute(() =>
             {
                 var serviceProvider = new ServiceProvider();
                 var context = CreateProjectContext(projectPath.Value());
+                
+                var configuration = appConfiguration.Value()??"Debug";
```
Comment (JunTaoLuo) [inline src/Microsoft.Extensions.CodeGeneration/Program.cs]: nit: spaces around `??`

File: test/Microsoft.Extensions.CodeGeneration.Core.FunctionalTest/TestApplicationEnvironment.cs [hunk 4]
```diff
@@ -42,7 +44,11 @@ public string ApplicationBasePath
         //{
         //    get { return _originalAppEnvironment.RuntimeFramework; }
         //}
-
+        
+        public string ApplicationConfiguration 
+        {
+            get { return _appConfiguration; }    
+        }
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.CodeGeneration.Core.FunctionalTest/TestApplicationEnvironment.cs]: nit: new line below

File: test/Microsoft.Extensions.CodeGeneration.Sources.Test/CommonUtilitiesTests.cs [hunk 5]
```diff
@@ -18,9 +18,15 @@ namespace Microsoft.Extensions.CodeGeneration
     public class CommonUtilitiesTests : TestBase
     {
         ICodeGenAssemblyLoadContext loadContext;
-
+        IApplicationEnvironment _environment;
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.CodeGeneration.Sources.Test/CommonUtilitiesTests.cs]: Why don't you set this in the TestBase to reduce duplication of code?

File: test/Microsoft.Extensions.CodeGeneration.Sources.Test/LibraryExporterTests.cs [hunk 6]
```diff
@@ -19,15 +19,16 @@ public class LibraryExporterTests : TestBase
 
         LibraryExporter _libraryExporter;
 
+
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.CodeGeneration.Sources.Test/LibraryExporterTests.cs]: nit: remove new line

File: test/Microsoft.Extensions.CodeGeneration.Sources.Test/LibraryExporterTests.cs [hunk 7]
```diff
@@ -19,15 +19,16 @@ public class LibraryExporterTests : TestBase
...
         public LibraryExporterTests() : base (@"..\TestApps\ModelTypesLocatorTestClassLibrary")
         {
-            _libraryExporter = new LibraryExporter(_projectContext);
+            _libraryExporter = new LibraryExporter(_projectContext, _environment);
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.CodeGeneration.Sources.Test/LibraryExporterTests.cs]: nit: spacing

**Comment by JunTaoLuo** [comment]
:shipit: after spacing comments

**Comment by prafullbhosale** [comment]
@JunTaoLuo have fixed the spacing issues in another PR  #178 
(Did not want to create a commit just for the spacing issues)

**Comment by JunTaoLuo** [comment]
:+1:

**Pull request merged**